src/zenml/cli/stack.py

```diff
@@ -1636,3 +1636,89 @@ def destroy(
                 f"Spec directory for stack '{stack_name}' successfully deleted."
             )
         cli_utils.declare(f"Stack '{stack_name}' successfully destroyed.")
+
+
+@stack.command(
+    "connect", help="Connect a service-connector to a stack components."
+)
+@click.argument("stack_name_or_id", type=str, required=False)
+@click.option(
+    "--connector",
+    "-c",
+    "connector",
+    help="The name, ID or prefix of the connector to use.",
+    required=False,
+    type=str,
+)
+@click.option(
+    "--resource-id",
+    "-r",
+    "resource_id",
+    help="The resource ID to use with the connector. Only required for "
+    "multi-instance connectors that are not already configured with a "
+    "particular resource ID.",
+    required=False,
+    type=str,
+)
+@click.option(
+    "--interactive",
+    "-i",
+    "interactive",
+    is_flag=True,
+    default=False,
+    help="Configure a service connector resource interactively.",
+    type=click.BOOL,
+)
+@click.option(
+    "--no-verify",
+    "no_verify",
+    is_flag=True,
+    default=False,
+    help="Skip verification of the connector resource.",
+    type=click.BOOL,
+)
+def connect_stack(
+    stack_name_or_id: Optional[str] = None,
+    connector: Optional[str] = None,
+    interactive: bool = False,
+    no_verify: bool = False,
+) -> None:
+    """Connect a service-connector to a stack components.
+
+    The stack command `connect` does not connect a service-connector to a
+    stack entity directly, as stack are only a collection of stack components
+    that can be updated or deleted. Instead, the `connect` command is used to
+    connect a service-connector to a stack component iteratively by going
+    through all stack components and trying to connect the service-connector
+    to each of them.
+
+    For this reason, the `connect` command does not save the state of the
+    connection to the stack component. Instead, the state is saved to the
+    service-connector resource. This means that the `connect` should be run
+    every time a stack component is added to a stack, if the same service-
+    connector is used for all stack components.
+
+    The advantage of this approach is that the service-connector resource
+    can be used to connect to multiple stack components in one go, without
+    having to run the `connect` command multiple times.
```

Comment:
IMO most of this is not relevant to users. If I understand it right, you could summarize it like this:

```suggestion
    """Connect a service-connector to all components of a stack.

    Note that this only connects the service-connector to the current 
    components of the stack and not to the stack itself, which means that 
    you need to rerun the command after adding new components to the stack.
```